Added highLevelDataSetup stage and re-ordered pipeline stages (#747)
hmcts/cnp-jenkins-library: README.md

The smoke tests are to be non-destructive (i.e. have no data impact, such as not creating accounts) and a subset of component level functional tests.

#### High level data setup

> Data requirements via high-level means like API calls can be satisfied under this stage. Most typical example is CCD definition files to be imported to target environments as part of a release activity, but data requirements of similar nature can be included under the same with our without a definition import accompanying the release. The satisfied data requirements can then be used within the same builds for automated testing, if available, in lower environments, and for actual live usage on production. By calling **enableHighLevelDataSetup()** in the related Jenkins file, HighLevelDataSetup stages in the opinionated pipeline are enabled to take effect.

Reviewer: I've re-worded it using I think more appropriate language, release activity is not something that is a concept in the Jenkins library, sentence per line markdown makes changes easier to review in future, and there were some grammatical issues which should be fixed by this change, feel free to adjust the below

```suggestion
This can be used to import data required for the application.
The most common example is CCD definition file, but data requirements of a similar nature can be included using the same functionality.
Smoke and functional tests in non-production environments will run after the import allowing automated regression testing of the change.

By adding `enableHighLevelDataSetup()` to the Jenkinsfile, `High Level Data Setup` stages will be added to the pipeline.
```

```
#!groovy

@Library("Infrastructure")

def type = "java"

> 

Reviewer: ```suggestion
```

def product = "rhubarb"

> 

Reviewer: ```suggestion
```

def component = "recipe-backend"

withPipeline(type, product, component) {
  enableHighLevelDataSetup()
}

```

The opinionated pipeline uses the following branch mapping to import definition files to different environments.

Branch | HighDataSetup Stage
--- | ---
`master` | `aat` then `prod`

> `PR` | `aat`

Reviewer: hmm many teams have their own instance of ccd on PR environments, maybe something for a future iteration but this likely needs to be more flexible, or disabled on PRs in some way.

Not a blocker

Reviewer 2: The current stage also allows PR builds to push to preview as well, but just takes a 1-liner to pre-suppress the aat url with any preview url.

The smoke tests are to be non-destructive (i.e. have no data impact, such as not creating accounts) and a subset of component level functional tests.

#### High level data setup

This can be used to import data required for the application.

> The most common example is CCD definition file, but data requirements of a similar nature can be included using the same functionality.

Reviewer: ```suggestion
The most common example is importing a CCD definition, but data requirements of a similar nature can be included using the same functionality.
```
